### PR TITLE
Add the Okta access point for the Okta service.

### PR DESCRIPTION
**Pull request by mdwn**
The Okta access point has been introduced for use by the Okta service.

Note: Cache implementation is here: https://github.com/gravitational/teleport/pull/23059

### CODE REVIEW

File: lib/auth/api.go [hunk 1]
```diff
@@ -721,6 +721,76 @@ type DiscoveryAccessPoint interface {
 	DeleteDatabase(ctx context.Context, name string) error
 }
 
+// ReadOktaAccessPoint is a read only API interface to be
+// used by an Okta component.
+//
+// NOTE: This interface must match the resources replicated in cache.ForOkta except for
```
Comment (ibeckermayer) [inline lib/auth/api.go:727]: I'm confused -- looking at the other PR, `ForOkta` appears to return a `Config` object. The relationship between this interface and that function is unclear.

File: lib/auth/api.go [hunk 2]
```diff
@@ -721,6 +721,76 @@ type DiscoveryAccessPoint interface {
...
+	GetOktaImportRule(ctx context.Context, name string) (types.OktaImportRule, error)
+
+	// ListOktaAssignments returns a paginated list of all Okta assignment resources.
+	ListOktaAssignments(context.Context, int, string) ([]types.OktaAssignment, string, error)
+
+	// GetOktaAssignmen treturns the specified Okta assignment resources.
+	GetOktaAssignment(ctx context.Context, name string) (types.OktaAssignment, error)
+}
+
+// OktaAccessPoint is a read only cache interface used by an Okta component.
```
Comment (ibeckermayer) [inline lib/auth/api.go:758]: It doesn't look like it's read only

**Review by ibeckermayer — APPROVED** [review]

File: lib/auth/api.go (hunk 2 above)
Comment (mdwn) [reply lib/auth/api.go:758]: Ahh whoops, I got a clarification after writing this comment and forgot to update it. Thank you!

File: lib/auth/api.go (hunk 1 above)
Comment (mdwn) [reply lib/auth/api.go]: I updated the comment -- let me know if the current version is clearer.

File: lib/auth/api.go [hunk 3]
```diff
@@ -1112,6 +1182,65 @@ func (w *DiscoveryWrapper) Close() error {
 	return trace.NewAggregate(err, err2)
 }
 
+type OktaWrapper struct {
```
[the remark this answers is not in the document]
Comment (mdwn) [reply lib/auth/api.go:1185]: Whoops. Added!

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**